test/e2e/framework.go

```diff
@@ -2892,6 +2892,14 @@ func (data *TestData) checkAntreaAgentInfo(interval time.Duration, timeout time.
 			// keep trying
 			return false, nil
 		}
+		//validate that the podRef in AntreaAgentInfo matches the name of the current Pod for the Node
+		pod, err := data.clientset.CoreV1().Pods(aai.PodRef.Namespace).Get(context.TODO(), aai.PodRef.Name, metav1.GetOptions{})
+		if err != nil {
```

Comment:
There could be two kinds of errors, NotFound and other transient errors. The former is kind of expected as it may happen that the new Pod hasn't updated AntreaAgentInfo yet (so the pod name in AntreaAgentInfo is stale), and we should keep retrying. However, returning non nil error to `PollImmediate` will make it fail immediately. Thus we should check if it's NotFound error like L2886 and only return `false, nil` in this case.